src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lectureUnit.service.ts

```diff
@@ -21,6 +21,14 @@ export class LectureUnitService {
     constructor(private httpClient: HttpClient, private attachmentService: AttachmentService, private exerciseService: ExerciseService) {}
 
     updateOrder(lectureId: number, lectureUnits: LectureUnit[]): Observable<HttpResponse<LectureUnit[]>> {
+        // need to remove participations from exercise units to prevent circular structure failure
+        lectureUnits = lectureUnits.map((lectureUnit) => {
+            if (lectureUnit.type === LectureUnitType.EXERCISE && (lectureUnit as ExerciseUnit).exercise) {
+                (lectureUnit as ExerciseUnit).exercise!.studentParticipations = undefined;
+                (lectureUnit as ExerciseUnit).exercise!.tutorParticipations = undefined;
+            }
```

Comment:
Casting three times doesn't look as performant as casting once, so I suggest doing something like:
```
if (lectureUnit.type !== LectureUnitType.EXERCISE) {
    return lectureUnit;
}
const exercise = (lectureUnit as ExerciseUnit).exercise;
exercise!.studentParticipations = undefinded;
exercise!.tutorParticipations = undefined;
return lectureUnit;
```

Reply:
Do not agree the new variable is imo more of a performance hit than giving the compiler type assertions

Reply:
I see, that is just compile time casting, so you're right.